src/crimson/admin/osd_admin.cc

```diff
@@ -265,4 +272,96 @@ class SeastarMetricsHook : public AdminSocketHook {
 };
 template std::unique_ptr<AdminSocketHook> make_asok_hook<SeastarMetricsHook>();
 
+
+static ghobject_t test_ops_get_object_name(
+  const OSDMap& osdmap,
+  const cmdmap_t& cmdmap)
+{
+  int64_t pool;
+  if (auto poolarg = cmd_getval<std::string>(cmdmap, "pool"); poolarg) {
+    pool = osdmap.lookup_pg_pool_name(*poolarg);
+    if (pool < 0 && std::isdigit((*poolarg)[0])) {
+      pool = std::atoll(poolarg->c_str());
+    }
+    if (pool < 0) {
+      // the return type of `fmt::format` is `std::string`
+      throw fmt::format("Invalid pool '{}'", *poolarg);
```

Comment:
nit, how about using `std::invalid_argument` or `crimson::osd::invalid_argument`? i am inclined to use classes inherited from `exception` for exceptions.

Reply:
also, could use
```c++
"Invalid pool '{}'"_format(*poolarg);
```
for compile time fmt parameter validation.